maven-extension/src/main/java/io/opentelemetry/maven/SpanRegistry.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.maven;
+
+import io.opentelemetry.api.trace.Span;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.component.annotations.Component;
+
+/** Holds the state of the spans in progress FIXME support multi module projects */
+@Component(role = SpanRegistry.class)
+public class SpanRegistry {
+
+  private final Map<MojoExecutionKey, Span> mojoExecutionKeySpanMap = new HashMap<>();
```

Comment:
Perhaps these aren't ConcurrentHashMap due to not supporting multi module yet

Reply:
I guess it was more a multi threading bug as I implemented multi module support a while ago 👼 good catch thanks!